ExampleMod/Content/Tiles/ExampleVanillaConversionTiles.cs

```diff
@@ -0,0 +1,255 @@
+using ExampleMod.Content.Items;
+using Microsoft.Xna.Framework;
+using Terraria;
+using Terraria.GameContent;
+using Terraria.ID;
+using Terraria.ModLoader;
+using Terraria.ObjectData;
+
+namespace ExampleMod.Content.Tiles
+{
+	// These three classes showcase how to create tiles that act as corruption/crimson/hallow versions of vanilla tiles.
+	// For this example, we will be making vanilla's desert fossil tiles convertible into the three spreading biomes
+	public class HallowedFossilTile : ModTile
+	{
+		public override void SetStaticDefaults() {
+			Main.tileSolid[Type] = true;
+			TileID.Sets.Hallow[Type] = true;
+			TileID.Sets.HallowBiome[Type] = true;
+			TileID.Sets.HallowBiomeSight[Type] = true;
+			TileID.Sets.HallowCountCollection[Type] = true;
+			DustType = DustID.Pearlsand;
+			AddMapEntry(new Color(157, 76, 152));
+
+			//We need to register a conversion from the vanilla desert fossil into our modded variants., so our custom code can be called when the game attempts to convert the vanilla tile
+			//We could register all three conversions in this class and reuse the same method for all three by checking for the conversionType there if we wanted to take up less space.
+			TileLoader.RegisterConversion(TileID.DesertFossil, BiomeConversionID.Hallow, ConvertToHallow);
+		}
+
+		public bool ConvertToHallow(int i, int j, int type, int conversionType) {
+
+			//This method is called whenever hallow biome conversion happens on a desert fossil tile, as per the RegisterConversion we called in SetStaticDefaults
+			//We don't need to check the type or the conversionType as we only registered one conversion with this method, but the same method could be reused for multiple conversion types or tiles
+
+			//We can use the ConvertTile utility method to change the fossil tile into our hallowed fossil tile, and it'll automatically handle tile frame updates and network syncing!
+			WorldGen.ConvertTile(i, j, Type);
+			return false;
+		}
+
+		//This code is called when the game attempts to convert our hallowed tile into a new biome
+		public override bool Convert(int i, int j, int conversionType) {
+			//Purification powder doesn't convert hallow tiles back into pure versions
+			if (conversionType == BiomeConversionID.PurificationPowder)
+				return false;
+			//Yellow (desert) solution also converts evil/hallowed tiles back into purity, so don't forget that check!
+			else if(conversionType == BiomeConversionID.Purity || conversionType == BiomeConversionID.Sand) {
```

Comment:
No need to use `else` statements when using `return` but consider using a `switch` statement